### PR TITLE
Added proofFormatMode flag and trivially used flag

### PR DESCRIPTION
**Pull request by scottviteri**

### CODE REVIEW

File: src/smt/proof_manager.cpp [hunk 1]
```diff
@@ -118,6 +121,10 @@ void PfManager::printProof(std::shared_ptr<ProofNode> pfn, Assertions& as)
   // TODO (proj #37) according to the proof format, print the proof node
   // leanPrinter(out, fp.get());
   std::ostream& out = *options::out();
+  if (options::proofFormatMode() == options::ProofFormatMode::LEAN) {
+    d_lpfpp->process(fp);
+    proof::leanPrinter(out, fp);
+  }
```
Comment (HanielB) [inline src/smt/proof_manager.cpp:127]: Note that you should be clang-formatting.

File: src/proof/lean/lean_printer.h [hunk 2]
```diff
@@ -14,23 +14,36 @@
...
+
+#include <iostream>
+#include "expr/proof_node.h"
+
+namespace CVC4 {
+namespace proof {
 
 enum LeanRule
 {
+  // in what format should I put these lean rules
```
Comment (HanielB) [inline src/proof/lean/lean_printer.h:28]: Let's try to follow the enumeration in `expr/proof_rule.h`. So an enum value for each lean of the Lean calculus.

**Comment by scottviteri** [comment]
Woops doesn't compile

File: src/proof/lean/lean_printer.h [hunk 3]
```diff
@@ -14,23 +14,37 @@
...
   // should print theorem statement
+  out << "hello world"
+      << "\n";
 
   // should traverse proof node, print each as a proof step, according to the
   // LEAN_RULE id in the LeanRule enum
   out << pfn;
 }
+
+#endif
```
Comment (HanielB) [inline src/proof/lean/lean_printer.h:48]: you should move this to the end of the file

File: src/proof/lean/lean_printer.h (hunk 2 above)
Comment (Lachnitt) [reply src/proof/lean/lean_printer.h:29]: Can we make this an enum class?

Comment (HanielB) [reply src/proof/lean/lean_printer.h:29]: Yes, we should. So it should change to `enum class LeanRule : uint32_t`.

**Pull request merged**